shared/lib/wp/support.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import qs from 'qs';
+
+export default function wpcomSupport( wpcom ) {
+	let supportUser = '';
+	let supportToken = '';
+
+	/**
+	 * Add the supportUser and supportToken to the query.
+	 * @param {Object}  params The original request params object
+	 * @return {Object}        The new query object with support data injected
+	 */
+	const addSupportData = function( params ) {
+		// Unwind the query string
+		let query = qs.parse( params.query );
+
+		// Inject the credentials
+		query.support_user = supportUser;
+		query._support_token = supportToken
+
+		return Object.assign( {}, params, {
+			query: qs.stringify( query )
+		} );
+	};
```

Comment:
Using `qs.parse` instead of a hacked-in version is smart; if the params change, we won't have to rewrite parts of `addSupportData`.